galaxy/tools/verify/interactor.py

```diff
@@ -287,29 +326,28 @@ def stage_data_async(self, test_data, history_id, tool_id, force_path_paste=Fals
                         "files_%d|url_paste" % i: "file://" + file_name
                     })
                 else:
-                    files["files_%s|file_data" % i] = open(file_name, 'rb')
+                    file_content = self.test_data_download(tool_id, file_name)
+                    files["files_%s|file_data" % i] = file_content
                 tool_input.update({
                     "files_%d|type" % i: "upload_dataset",
                 })
             name = test_data['name']
         else:
-            file_name = self.test_data_path(tool_id, fname)
-            name = test_data.get('name', None)
-            if not name:
-                name = os.path.basename(file_name)
-
+            name = fname
             tool_input.update({
                 "files_0|NAME": name,
                 "files_0|type": "upload_dataset",
+
             })
             files = {}
             if force_path_paste:
                 tool_input.update({
                     "files_0|url_paste": "file://" + file_name
```

Comment:
Similar problem here, `file_name` is now undefined in this context.

Reply:
Fixed I think. Still not tested...